tests/testthat/test-parallel.R

```diff
@@ -0,0 +1,132 @@
+})
+
+test_that("enable future parallelism", {
+  skip_if_not_installed("future")
+
+  svm_spec <- svm_rbf(mode = "classification")
+  ctrl_no <- control_grid(allow_par = FALSE)
+  ctrl_java <- control_grid(pkgs = "rJava")
+
+  library(future)
+
+  # ----------------------------------------------------------------------------
+  # sequential; not enough workers
+
+  expect_equal(tune:::choose_framework(), "sequential")
+  expect_snapshot(tune:::choose_framework(verbose = TRUE))
+
+  # ----------------------------------------------------------------------------
+  # parallel
+
+  plan(multisession(workers = 2))
+  expect_equal(tune:::choose_framework(), "future")
+  expect_snapshot(tune:::choose_framework(verbose = TRUE))
+  expect_equal(tune:::choose_framework(svm_spec), "future")
+
+  # ----------------------------------------------------------------------------
+  # sequential due to restrictions
+
+  expect_equal(tune:::choose_framework(control = ctrl_no), "sequential")
+  expect_snapshot(tune:::choose_framework(control = ctrl_no, verbose = TRUE))
+
+  expect_equal(tune:::choose_framework(control = ctrl_java), "sequential")
+  expect_snapshot(tune:::choose_framework(control = ctrl_java, verbose = TRUE))
+
+  plan(multisession(workers = 1))
+
+  expect_equal(tune:::choose_framework(), "sequential")
+  expect_snapshot(tune:::choose_framework(verbose = TRUE))
+
+  plan(sequential)
+  detach("package:future", character.only = TRUE)
+
+  expect_equal(tune:::choose_framework(), "sequential")
+  expect_snapshot(tune:::choose_framework(verbose = TRUE))
+})
+
+test_that("enable mirai parallelism", {
+  skip_if_not_installed("mirai", minimum_version = "2.4.0")
+
+  svm_spec <- svm_rbf(mode = "classification")
+  ctrl_no <- control_grid(allow_par = FALSE)
+  ctrl_java <- control_grid(pkgs = "rJava")
+
+  library(mirai)
+
+  # ----------------------------------------------------------------------------
+  # sequential; not enough workers
+
+  expect_equal(tune:::choose_framework(), "sequential")
+  expect_snapshot(tune:::choose_framework(verbose = TRUE))
+
+  # ----------------------------------------------------------------------------
+  # parallel
+
+  daemons(2)
+
+  expect_equal(tune:::choose_framework(), "mirai")
+  expect_snapshot(tune:::choose_framework(verbose = TRUE))
+  expect_equal(tune:::choose_framework(svm_spec), "mirai")
+
+  # ----------------------------------------------------------------------------
+  # sequential due to restrictions
+
+  expect_equal(tune:::choose_framework(control = ctrl_no), "sequential")
+  expect_snapshot(tune:::choose_framework(control = ctrl_no, verbose = TRUE))
+
+  expect_equal(tune:::choose_framework(control = ctrl_java), "sequential")
+  expect_snapshot(tune:::choose_framework(control = ctrl_java, verbose = TRUE))
+
+  daemons(1)
+
+  expect_equal(tune:::choose_framework(), "sequential")
+  expect_snapshot(tune:::choose_framework(verbose = TRUE))
+
+  daemons(0)
+  ## Give daemons a chance to shutdown
```

Comment:
I was seeing issues where some workers were still up and around when resetting `daemons`. I was getting errors `daemons already set for 'default' compute profile`. 

I found this in the future.mirai repo. 

Is there a better way to teardown the workers?